src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.Split.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System
+{
+    public static partial class MemoryExtensions
+    {
+        public static SpanSplitEnumerator<char> Split(this ReadOnlySpan<char> span)
+            => new SpanSplitEnumerator<char>(span, ' ');
+
+        public static SpanSplitEnumerator<char> Split(this ReadOnlySpan<char> span, char separator)
+            => new SpanSplitEnumerator<char>(span, separator);
+
+        public static SpanSplitEnumerator<char> Split(this ReadOnlySpan<char> span, string separator)
+            => new SpanSplitEnumerator<char>(span, separator);
+    }
+
+    public ref struct SpanSplitEnumerator<T> where T : IEquatable<T>
+    {
+        private readonly ReadOnlySpan<T> _sequence;
+        private readonly ReadOnlySpan<T> _separators;
+        private readonly T _separator;
+        private readonly bool _isSequence;
+        private readonly int _separatorLength;
```

Comment:
This could be replaced with a cleaner:
```
private int SeparatorLength => _isSequence ? _separators.Length : 1;
```
However, according to my benchmarks it will incur a 3% perf penalty on the tested scenario.
What would be preferred in this case?